osu.Game.Rulesets.Osu/Difficulty/OsuPerformanceCalculator.cs

```diff
@@ -138,8 +139,9 @@ private double computeSpeedValue()
                                  (totalHits > 2000 ? Math.Log10(totalHits / 2000.0) * 0.5 : 0.0);
             speedValue *= lengthBonus;
 
-            // Penalize misses exponentially. This mainly fixes tag4 maps and the likes until a per-hitobject solution is available
-            speedValue *= Math.Pow(0.97, countMiss);
+            // Penalize misses by assessing # of misses relative to the total # of objects. Default a 3% reduction for any # of misses.
+            if (countMiss > 0)
+                speedValue *= 0.97 * Math.Pow(1 - Math.Pow((double)countMiss / (double)totalHits, 0.775), countMiss);
```

Comment:
```suggestion
                speedValue *= 0.97 * Math.Pow(1 - Math.Pow((double)countMiss / totalHits, 0.775), countMiss);
```

Reply:
done, but note that I did move that previous existing ^.75 miss penalty from aim to speed to better balance misses on stream maps. this is what i believe I wanted originally but I must've accidently switched them. I was exhausted by the end of yesterday. the ^.75 to misscount just makes misses penalize a little less on speed, which makes sense since the skill is heavily correlated with accuracy and less so on misses.